src/models/plateau.rs

```diff
@@ -0,0 +1,169 @@
+use crate::models::{piece, point};
+
+use piece::Piece;
+use point::Point;
+use std::fmt;
+
+const PLAYER1: char = '0';
+const PLAYER2: char = 'X';
+const EMPTY: char = '.';
+
+#[derive(Debug, Clone, PartialEq)]
+enum Cell {
+	Player1,
+	Player2,
+	Empty,
+}
+
+pub enum Player {
+	Player1,
+	Player2,
+}
+
+#[derive(Debug)]
+pub struct Plateau {
+	pub width: u32,
+	pub height: u32,
+	cells: Vec<Cell>,
+}
+
+impl Plateau {
+	pub fn new(
+		width: u32,
+		height: u32,
+		player1: &Point,
+		player2: &Point,
+	) -> Result<Plateau, String> {
+		let mut plateau = Plateau {
+			width,
+			height,
+			cells: vec![Cell::Empty; (width * height) as usize],
+		};
+
+		match plateau.is_in_bounds(player1) {
+			true => plateau.set(player1, Cell::Player1),
+			false => return Err(String::from("Player1 out of bounds")),
+		};
+
+		match plateau.is_in_bounds(player2) {
+			true => plateau.set(player2, Cell::Player2),
+			false => return Err(String::from("Player2 out of bounds")),
+		};
+
+		Ok(plateau)
+	}
+
+	pub fn is_in_bounds(&self, p: &Point) -> bool {
+		p.x < self.width && p.y < self.height
```

Comment:
We'll have to allow placement with negative values, as pieces can be placed beyond the left and upper boundaries while remaining valid.

Reply:
Will raise this as a separate issue to be implemented.